web-runtime/src/lib.rs

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2020 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#![warn(rust_2018_idioms)]
+#![feature(get_mut_unchecked)]
+#![feature(new_uninit)]
+#![feature(stmt_expr_attributes)]
+#![deny(
+    dead_code,
+    nonstandard_style,
+    unused_imports,
+    unused_mut,
+    unused_variables,
+    unused_unsafe,
+    unreachable_patterns
+)]
+
+pub(crate) mod marine_js;
+mod engine;
+mod errors;
+mod misc;
+mod module;
+mod faas;
+
+use std::cell::RefCell;
+use std::collections::HashMap;
+use wasm_bindgen::prelude::*;
+use module::MModule;
+
+pub use engine::MModuleInterface;
+pub use engine::Marine;
+pub use errors::MError;
+pub use module::IValue;
+pub use module::IRecordType;
+pub use module::IFunctionArg;
+pub use module::IType;
+pub use module::MRecordTypes;
+pub use module::MFunctionSignature;
+pub use module::from_interface_values;
+pub use module::to_interface_value;
+pub use wasmer_it::IRecordFieldType;
+
+use crate::faas::FluenceFaaS;
+use marine_rs_sdk::CallParameters;
+
+use once_cell::sync::Lazy;
+
+use std::str::FromStr;
+pub use wasmer_it::ne_vec;
+
+pub(crate) type MResult<T> = std::result::Result<T, MError>;
+
+static MINIMAL_SUPPORTED_IT_VERSION: Lazy<semver::Version> = Lazy::new(|| {
+    semver::Version::from_str("0.20.0").expect("invalid minimal sdk version specified")
+});
+
+// These locals intended for check that set versions are correct at the start of an application.
+thread_local!(static MINIMAL_SUPPORTED_IT_VERSION_CHECK: &'static semver::Version = Lazy::force(&MINIMAL_SUPPORTED_IT_VERSION));
+thread_local!(static MODULES: RefCell<Option<FluenceFaaS>> = RefCell::new(None));
+thread_local!(static INSTANCE: RefCell<Option<JsValue>> = RefCell::new(None));
+
+/// Return minimal support version of interface types.
+pub fn min_it_version() -> &'static semver::Version {
+    Lazy::force(&MINIMAL_SUPPORTED_IT_VERSION)
+}
+
+#[wasm_bindgen]
```

Comment:
I suggest to decouple all API functions to a separate file to make `lib.rs` more informative.